content/300-guides/300-upgrade-guides/200-upgrading-versions/030-upgrading-to-prisma-4/index.mdx

```diff
@@ -0,0 +1,170 @@
+
+In Prisma 4, the `extendedIndexes` Preview feature will now become generally available. This includes the following index configuration options:
+
+- Length configuration of indexes, unique constraints and primary key constraints for MySQL (in Preview in versions `3.5.0` and later)
+- Sort order configuration of indexes, unique constraints and primary key constraints (in Preview in versions `3.5.0` and later)
+- New index types for PostgreSQL: Hash (in Preview in versions `3.6.0` and later) and GIN, GiST, SP-GiST and BRIN (in Preview in versions `3.14.0` and later)
+- Index clustering for SQL Server (in Preview in versions `3.13.0` and later)
+
+#### Length configuration (MySQL)
+
+#### Sort order configuration
+
+#### New index types (PostgreSQL)
+
+#### Index clustering (SQL Server)
+
+### Scalar list defaults
+
+For database connectors that support scalar lists (PostgreSQL, CockroachDB and MongoDB), Prisma 4 introduces the ability to set a default value in your Prisma schema file using the `@default` attribute:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma highlight=3;normal
+model User {
+  id             Int      @id @default(autoincrement())
+  posts          Post[]
+  favoriteColors String[] @default(["red", "yellow", "purple"])
+}
+```
+
+</tab>
+<tab>
+
+```prisma highlight=3;normal
+model User {
+  id             String   @id @default(auto()) @map("_id") @db.ObjectId
+  posts          Post[]
+  favoriteColors String[] @default(["red", "yellow", "purple"])
+}
+```
+
+</tab>
+</TabbedContent>
+
+This is a breaking change if you previously had defaults defined for scalar lists at the database level.
+
+### Explicit `@unique` constraints on one-to-one relations
+
+When using one-to-one relations in Prisma 4, you will need to explicitly add the `@unique` attribute on the relation scalar field. For example, for this one-to-one relation between a `User` and a `Profile` model, you will need to add the `@unique` attribute to the `profileId` field:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  id        Int      @id @default(autoincrement())
+  profile   Profile? @relation(fields: [profileId], references: [id])
+  profileId Int?     @unique // <-- include this explicitly
+}
+
+model Profile {
+  id   Int   @id @default(autoincrement())
+  user User?
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+model User {
+  id        String   @id @default(auto()) @map("_id") @db.ObjectId
+  profile   Profile? @relation(fields: [profileId], references: [id])
+  profileId String?  @unique @db.ObjectId // <-- include this explicitly
+}
+
+model Profile {
+  id   String @id @default(auto()) @map("_id") @db.ObjectId
+  user User?
+}
+```
+
+</tab>
+</TabbedContent>
+
+### Removing `references` syntax for implicit many-to-many relations
```

Comment:
We can be stronger: "Forbidding" or "Disallowing" would also fit.

Reply:
Changed to Disallow